src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/exceptions/ValidationErrorHelper.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * The MIT License
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.exceptions;
+
+import org.niis.xroad.restapi.openapi.model.CodeWithMetadata;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+
+import java.util.Collections;
+
+/**
+ * Helper class for transforming Spring validation errors into error dtos.
+ * Used from {@link ExceptionTranslator} and {@link SpringInternalExceptionHandler}
+ */
+public class ValidationErrorHelper {
+    public static final String VALIDATION_FAILURE_ERROR = "validation_failure";
+
+    /**
+     * Create DeviationAware error code & metadata from given org.springframework.validation.Errors.
+     * Error code = {@link ValidationErrorHelper#VALIDATION_FAILURE_ERROR}, metadata is the
+     * String representation of {@link Errors}
+     */
+    public CodeWithMetadata createError(Errors validationErrors) {
```

Comment:
True. I avoided statics on purpose, since I have this uneasy feeling with static utility classes. I always fear I might end up needing to mock those in tests, or override the util class behavior. Something about static methods always makes my skin crawl. Too many bad experiences. 

I asked https://gofore.slack.com/archives/CCA0KKAVD/p1586356799085900 to help me find the light. 

That being said, this feels like a good example of utility methods that a) won't change b) won't need to be mocked c) will only ever need one implementation so if we use static utility classes this is a good fit for them. And we do use static utility classes.

I will either change this to static util class, or if not that then a Spring singleton bean. Now it's kind of in-between.

I guess I still hold my opinion that static methods / static utility classes have lots of potential for causing troubles and use of them should be minimized.

Reply:
Based on feedback from the discussion thread, I am now convinced that "beans always" is a valid default strategy and static utility classes should be limited to rare cases where you can be certain that you will never need to mock the methods or use inheritance. It is subjective, sure.

So I changed this one to a spring bean since I felt I might want to mock these in some test. There's a bit of boilerplate bloat as a result, which is the price for this choice.